### ISSUE TITLE
Crash in S6Exporter::Save

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#4436]
**OS:** Windows 10
**Commit/Build:** f538d2f

Crash dump from Whiskey server, an odd one.

- [x] Multiplayer?

```
Thread 0 (crashed)
 0  openrct2.dll!_invalid_parameter [invalid_parameter.cpp : 93 + 0x0]
    eip = 0x7255a14e   esp = 0x000df9c0   ebp = 0x000df9dc   ebx = 0x00000000
    esi = 0x01515798   edi = 0x00000000   eax = 0x000df6e0   ecx = 0x00000000
    edx = 0x00000000   efl = 0x00200212
    Found by: given as instruction pointer in context
 1  openrct2.dll!_invalid_parameter_noinfo [invalid_parameter.cpp : 97 + 0xc]
    eip = 0x7255a198   esp = 0x000df9e4   ebp = 0x000dfa1c
    Found by: call frame info
 2  openrct2.dll!_fread_nolock_s [fread.cpp : 81 + 0x17]
    eip = 0x7255e2b7   esp = 0x000df9fc   ebp = 0x000dfa1c
    Found by: call frame info
 3  openrct2.dll!fread_s [fread.cpp : 56 + 0x12]
    eip = 0x7255e4de   esp = 0x000dfa24   ebp = 0x000dfa64
    Found by: call frame info
 4  openrct2.dll!fread [fread.cpp : 233 + 0x13]
    eip = 0x7255e46c   esp = 0x000dfa6c   ebp = 0x000dfa80
    Found by: call frame info
 5  openrct2.dll!SDL_WriteU8_REAL + 0x1c9
    eip = 0x723926d9   esp = 0x000dfa88   ebp = 0x000dfaa0
    Found by: call frame info
 6  openrct2.dll!S6Exporter::Save(SDL_RWops *,bool) [s6exporter.cpp : 218 + 0xf]
    eip = 0x722a1b71   esp = 0x000dfaa8   ebp = 0x000dfaf8
    Found by: previous frame's frame pointer
 7  openrct2.dll!scenario_save_network [s6exporter.cpp : 517 + 0xa]
    eip = 0x722a27c4   esp = 0x000dfb00   ebp = 0x000dfb30
    Found by: call frame info
 8  openrct2.dll!Network::Server_Send_MAP(NetworkConnection *) [network.cpp : 1010 + 0xa]
    eip = 0x72263d3e   esp = 0x000dfb38   ebp = 0x000dfbac
    Found by: call frame info
 9  openrct2.dll!Network::Server_Client_Joined(char const *,std::basic_string<char,std::char_traits<char>,std::allocator<char> > const &,NetworkConnection &) [network.cpp : 1528 + 0x8]
    eip = 0x722669bb   esp = 0x000dfbb4   ebp = 0x000dfcd0
    Found by: call frame info
10  openrct2.dll!Network::Server_Handle_AUTH(NetworkConnection &,NetworkPacket &) [network.cpp : 1607 + 0x14]
    eip = 0x72266e29   esp = 0x000dfcd8   ebp = 0x000dfd54
    Found by: call frame info
11  openrct2.dll!Network::ProcessPacket(NetworkConnection &,NetworkPacket &) [network.cpp : 1280 + 0x9]
    eip = 0x72265be2   esp = 0x000dfd5c   ebp = 0x000dfed8
    Found by: call frame info
12  openrct2.dll!Network::ProcessConnection(NetworkConnection &) [network.cpp : 1248 + 0xa]
    eip = 0x72265afe   esp = 0x000dfd70   ebp = 0x000dfed8
    Found by: call frame info
13  openrct2.dll!Network::UpdateServer() [network.cpp : 387 + 0xb]
    eip = 0x72261a0b   esp = 0x000dfe8c   ebp = 0x000dfed8
    Found by: call frame info
14  openrct2.dll!rct2_update [rct2.c : 449 + 0xf]
    eip = 0x7222e830   esp = 0x000dfea0   ebp = 0x000dfed8
    Found by: call frame info
15  openrct2.dll!openrct2_loop [openrct2.c : 473 + 0x5]
    eip = 0x7222d9b2   esp = 0x000dfeb0   ebp = 0x000dfed8
    Found by: call frame info
16  openrct2.dll!openrct2_launch [openrct2.c : 337 + 0x5]
    eip = 0x7222d4f3   esp = 0x000dfee0   ebp = 0x000dff80
    Found by: call frame info
17  openrct2.dll!StartOpenRCT [windows.c : 134 + 0x5]
    eip = 0x7229b0ed   esp = 0x000dfee4   ebp = 0x000dff80
    Found by: call frame info
18  openrct2.exe + 0x69e4
    eip = 0x004069e4   esp = 0x000dfee8   ebp = 0x000dff80
    Found by: call frame info
19  kernel32.dll + 0x138f4
    eip = 0x76b638f4   esp = 0x000dff88   ebp = 0x000dff94
    Found by: previous frame's frame pointer
20  ntdll.dll + 0x65de3
    eip = 0x77965de3   esp = 0x000dff9c   ebp = 0x000dffdc
    Found by: previous frame's frame pointer
21  ntdll.dll + 0x65dae
    eip = 0x77965dae   esp = 0x000dffe4   ebp = 0x000dffec
    Found by: previous frame's frame pointer
```

**Dump file** (includes debug symbols already for your convenience)
[13633ad0-9d50-4d7a-bac0-381294470cec.zip](https://github.com/OpenRCT2/OpenRCT2/files/473957/13633ad0-9d50-4d7a-bac0-381294470cec.zip)

**Comment by janisozaur** [comment]
Previously the server used to mysteriously think the map was 2GiB in size (see https://github.com/OpenRCT2/OpenRCT2/pull/4189) perhaps, there's still something odd around that.

**Comment by janisozaur** [comment]
Most likely fixed with b90f27a9718f9a158448f489961e4fb3850a057c
